SpatialGDK/Source/SpatialGDK/Public/SpatialView/ViewCoordinator.h

```diff
@@ -52,6 +52,19 @@ class ViewCoordinator
 	const FString& GetWorkerId() const;
 	Worker_EntityId GetWorkerSystemEntityId() const;
 
+	template <class T>
+	TOptional<T> GetComponent(Worker_EntityId EntityId) const
```

Comment:
Do we want to always be deserialising this at the point of reading the component?
In most places if we need to store data in some other format we've built that up in the system that deals with that component. For a large component this will be expensive and it enforces use of the component wrapper types that we aren't sure we want to keep around.

Reply:
I see this more as an easy was of getting your `T` component if you just need it as a one-off or as an initial value. Can definitely see how this can be abused, but not so sure on whether it's a problem.

Reply:
Discussed offline - will move this in a separate template, deserializing components really isn't something ViewCoordinator should be doing.